oak/proto/permissions.proto

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.permissions;
+
+// A PermissionsConfiguration provides a declarative description of the features that are permitted
+// for an Oak application.
+//
+// The permissions are applied dynamically at runtime, when creating the Nodes.
+message PermissionsConfiguration {
+  // Whether to enable the use of a GrpcServerNode as the front-door to the application.
+  bool grpc_front_door = 1;
+
+  // Whether to enable the use of an HttpServerNode as the front-door to the application.
+  bool http_front_door = 2;
+
+  // Whether to allow application logging. If true, allows creating a LogNode for the application.
+  bool logging = 3;
+
+  // Whether to allow monitoring (for the Runtime and the application). If true, the monitoring
+  // auxiliary service will be started.
+  bool monitoring = 4;
+
+  // Whether to allow introspection. If true, the introspection auxiliary service will be started.
+  bool introspection = 5;
+
+  // List of the allowed domain names or IP addresses that the application can connect to. Only
+  // GrpcClientNodes that are configured to connect to servers in this list can be created.
+  repeated string grpc_client_domains = 6;
+
+  // List of allowed roughtime servers that the application can connect to.
+  repeated string roughtime_servers = 7;
+
+  // Map from Wasm module names to the list of files they are allowed to access.
+  map<string, FileConfig> permitted_files = 8;
```

Comment:
I am not sure how this will be used; this configuration should be agnostic of the oak application configuration, so how would someone configure the appropriate wasm module names?

Reply:
I am struggling to think of a use case for limiting access to the contents of a file to only a specific WASM node. Do you have some use cases in mind?

Reply:
@conradgrobler I don't have a specific use case. But the idea was to have something similar to volumes in k8s deployment configs, where volumes are defined, and each container specifies which of the volumes it wants to mount to. 
@tiziano88 If not referencing the wasm modules, I don't know how else we can make a restriction on the files that a wasm module can use. Why does this config have to be agnostic of the oak application config? Or maybe I have misunderstood the requirement here.

Reply:
This is a config that someone who wants to run applications would write once and for all to determine what they are comfortable with, before having an actual application to deploy. It is meant to enable or disable features of the runtime that are not expected to be used in a particular environment. For instance, this would be something that we would show during a launch review, to convince the security team that we are not going to connect to arbitrary websites etc. But this audience would not know much / anything about the oak application that will run on the runtime.

Additionally, since files need to be explicitly passed to the runtime anyways, I don't think this level of defense in depth is particularly useful, it is just going to be a copy of the same list of files that we would pass there.

Reply:
Removed, as discussed.